memory.c

```diff
@@ -27,13 +27,22 @@
 
 void memory_protect(void)
 {
+	// Reference STM32F205 Flash programming manual revision 5 http://www.st.com/resource/en/programming_manual/cd00233952.pdf
+	// Section 2.6 Option bytes
 	//                     set RDP level 2                   WRP for sectors 0 and 1
 	if ((((*OPTION_BYTES_1) & 0xFFFF) == 0xCCFF) && (((*OPTION_BYTES_2) & 0xFFFF) == 0xFFFC)) {
 		return; // already set up correctly - bail out
 	}
-	flash_unlock_option_bytes();
```

Comment:
Good find, but I don't think keeping this has any side effects. I would like to keep that because it is more explicit.